test/data/data_tx_test.go

```diff
@@ -0,0 +1,527 @@
+		verKey3 := resp.GetResponse().GetMetadata().GetVersion()
+		require.NotNil(t, verKey3)
+
+		resp, err = s.QueryData(t, "db2", "key4", "alice")
+		require.NoError(t, err)
+		verKey4 := resp.GetResponse().GetMetadata().GetVersion()
+		require.NotNil(t, verKey4)
+
+		dataTx := &types.DataTx{
+			MustSignUserIds: []string{
+				"alice",
+				"bob",
+			},
+			TxId: uuid.New().String(),
+			DbOperations: []*types.DBOperation{
+				{
+					DbName: "db1",
+					DataReads: []*types.DataRead{
+						{
+							Key:     "key3",
+							Version: verKey3,
+						},
+					},
+					DataWrites: []*types.DataWrite{
+						{
+							Key:   "key3",
+							Value: []byte("key3value2"),
+							Acl: &types.AccessControl{
+								ReadWriteUsers:     map[string]bool{"alice": true, "bob": true},
+								SignPolicyForWrite: types.AccessControl_ALL,
+							},
+						},
+						{
+							Key:   "key5",
+							Value: []byte("key5constantValue"),
+							Acl: &types.AccessControl{
+								ReadUsers: map[string]bool{
+									"alice": true,
+								},
+							},
+						},
+					},
+					DataDeletes: []*types.DataDelete{
+						{
+							Key: "key1",
+						},
+					},
+				},
+				{
+					DbName: "db2",
+					DataReads: []*types.DataRead{
+						{
+							Key:     "key4",
+							Version: verKey4,
+						},
+					},
+					DataWrites: []*types.DataWrite{
+						{
+							Key:   "key4",
+							Value: []byte("key4value2"),
+							Acl: &types.AccessControl{
+								ReadWriteUsers:     map[string]bool{"alice": true, "bob": true},
+								SignPolicyForWrite: types.AccessControl_ALL,
+							},
+						},
+						{
+							Key:   "key6",
+							Value: []byte("key6constantValue"),
+							Acl: &types.AccessControl{
+								ReadUsers: map[string]bool{
+									"bob": true,
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		aliceSigner, err := s.Signer("alice")
+		require.NoError(t, err)
+
+		bobSigner, err := s.Signer("bob")
+		require.NoError(t, err)
+
+		_, err = s.SubmitTransaction(t, constants.PostDataTx, &types.DataTxEnvelope{
+			Payload: dataTx,
+			Signatures: map[string][]byte{
+				"alice": testutils.SignatureFromTx(t, aliceSigner, dataTx),
+				"bob":   testutils.SignatureFromTx(t, bobSigner, dataTx),
+			},
+		})
+		require.NoError(t, err)
+
+		resp, err = s.QueryData(t, "db1", "key3", "bob")
+		require.NoError(t, err)
+		require.Equal(t, []byte("key3value2"), resp.GetResponse().GetValue())
+
+		resp, err = s.QueryData(t, "db1", "key5", "alice")
+		require.NoError(t, err)
+		require.Equal(t, []byte("key5constantValue"), resp.GetResponse().GetValue())
+
+		resp, err = s.QueryData(t, "db2", "key4", "alice")
+		require.NoError(t, err)
+		require.Equal(t, []byte("key4value2"), resp.GetResponse().GetValue())
+
+		resp, err = s.QueryData(t, "db2", "key6", "bob")
+		require.NoError(t, err)
+		require.Equal(t, []byte("key6constantValue"), resp.GetResponse().GetValue())
+	})
+
+	t.Run("read-write-delete to a single database", func(t *testing.T) {
```

Comment:
All negative test cases names should be prefixed with the like of:
"invalid: read-write-delete to a single database"

Reply:
Done